### DISCUSSION TITLE
Nested collection editing inside another collection

### DISCUSSION THREAD

**Discussion by LoydAndrew** [payloadcms/payload#4106] [Feature Requests & Ideas]
Thank you for your work!
Is there a way to include and edit nested collection within another collection - so if you have books collection and authors collection and want to create a book you can write down and create  a new author within that form, rather than create it in a separate admin view for authors collection?
e.g
```js
const Books: CollectionConfig = {
  access: {
    delete: () => false,
    read: () => true,
  },
  admin: {
    useAsTitle: 'title',
  },
  fields: [
    {
      name: 'title',
      type: 'text',
    },
     ...Authors
  ],
  slug: 'categories',
}

```

**Comment by id3er0** [comment]
For this case I use a set of fields `src/fields/author.ts`:

```
import { Field } from 'payload/types';

const authorFields: Field[] = [
  {
    name: 'firstName',
    type: 'text',
  },
  {
    name: 'lastName',
    type: 'text',
  },
]
```

And import it in collections:

```
import { authorFields } from '@/fields';

const Books: CollectionConfig = {
  <...>
  fields: [
    {
      name: 'title',
      type: 'text',
    },
    ...authorFields,
  ],
}
```

```
import { authorFields } from '@/fields';

const Authors: CollectionConfig = {
  <...>
  fields: [
    {
      name: 'someField',
      type: 'textarea',
    },
    ...authorFields,
  ],
}
```

**Reply by LoydAndrew** [reply]
It's just gonna act like a set of fields, what I want to achieve is a collection within another collection that's gonna be saved in respective tables (author fields in Author table, books fields in Book table) in database after any manipulations.
